### DISCUSSION TITLE
How to add a customize message

### DISCUSSION THREAD

**Discussion by hudan8080808** [bluenviron/gomavlib#28] [Q&A]

**Comment by hudan8080808** [comment]
add a customize message,but it always tips a error called "duplicate ID"

**Comment by aler9** [comment]
Hello, to add a custom message you have to find an ID that is not in use from the dialect that you're using. Please refer to this examples for instructions on how to define a custom dialect or a custom message:

https://github.com/aler9/gomavlib/blob/main/examples/dialect-custom/main.go

**Comment by hudan8080808** [comment]
Thank you for your reply. But  connecting to the ardupolit, the calculated CRC value is incorrect,I want to know the CRC_EXTRA should be setting whether or not ,and if yes ,how to set?

**Comment by aler9** [comment]
Hello, CRC_EXTRA is computed automatically from the message struct. If the CRC_EXTRA of the Go message doesn't match the one of the Ardupilot message, it means that the Go message definition is different than the Ardupilot one.

You have to write a .xml containing your dialect, and then:
* use it into Ardupilot
* use it to generate Go message definitions by using the Dialect importer, as described in the README:
   https://github.com/aler9/gomavlib#dialect-generation
